### PR TITLE
Run clippy and build docs on CI

### PR DESCRIPTION
**Pull request by rzadp**

### CODE REVIEW

**Review by Bullrich — APPROVED** [review]

**Comment by alvicsam** [comment]
I'd also add concurrency section after `on:` to skip previous runs

```yml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

**Comment by rzadp** [comment]
> I'd also add concurrency section after `on:` to skip previous runs

Good call, done.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,6 +11,9 @@ on:
 jobs:
   ci:
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+      cancel-in-progress: true
```
Comment (alvicsam) [inline .github/workflows/ci.yml:16]:
```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
jobs:
  ci:
    runs-on: ubuntu-latest
```
Not sure that it works on the jobs level

Comment (rzadp) [reply .github/workflows/ci.yml:16]: It does: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency

**Review by alvicsam — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alvicsam) [reply .github/workflows/ci.yml:16]: ah, okay, since there is only one job there is not much difference where to put it

**Pull request merged**